cluster/tenant-service-accounts.go

```diff
@@ -0,0 +1,124 @@
+// This file is part of MinIO Kubernetes Cloud
+// Copyright (c) 2019 MinIO, Inc.
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+package cluster
+
+import (
+	"errors"
+
+	uuid "github.com/satori/go.uuid"
+)
+
+type ServiceAccount struct {
+	ID          uuid.UUID
+	Name        string
+	Description *string
+	AccessKey   string
+}
+
+// AddUser adds a new user to the tenant's database
+func AddServiceAccount(ctx *Context, tenantShortName string, name string, description *string) (*ServiceAccountCredentials, error) {
+
+	// Add parameters to query
+	serviceAccountID := uuid.NewV4()
+	query := `INSERT INTO
+				service_accounts ("id","name","description","sys_created_by")
+			  VALUES
+				($1,$2,$3,$4)`
+	tx, err := ctx.TenantTx()
+	if err != nil {
+		return nil, err
+	}
+	stmt, err := tx.Prepare(query)
+	if err != nil {
+		ctx.Rollback()
+		return nil, err
+	}
```

Comment:
can we implement her using defer?
like:
```
defer func() {
		if err != nil {
			ctx.Rollback()
			return
		}
		// if no error happened to this point commit transaction
		err = ctx.Commit()
	}()
```

Reply:
seems like I would have to return named parameters, but since the service account is generated by an inner function I cannot use this style of deferring